language/vm/vm-runtime/src/system_txn/block_metadata_processor.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{
```

Comment:
Do we want a separate file for this? The logic is pretty short/straightforward; the majority of the file is copying over the imports from the runtime.

Reply:
I do want to keep them separate as we have more system transactions in the future. I was thinking about having one file for each of different type of system transaction.